(Data de Entrega: 08/04  2º Feira)  
# Atividade Avaliativa

https://www-bcf.usc.edu/~gareth/ISL/ISLR%20First%20Printing.pdf 

Pág. 121 -> 8 (fazer todo), 9 (até letra d), 10 (fazer todo), 13 (fazer todo) e 15 (fazer todo). 
_________________________________________________________________________________________________

- 8. Esta questão envolve o uso de regressão linear simples no conjunto de dados [Auto].  

(a) Use a função lm() para executar uma regressão linear simples com mpg como a resposta e a horsepower 
como o preditor. 
         Use a função summary() para imprimir os resultados. Comente a saída.
Por exemplo:  

i. Existe uma relação entre o preditor e a resposta?

ii. Quão forte é a relação entre o preditor e a resposta?  

iii. A relação entre o preditor e a resposta é positiva ou negativa?  

iv. Qual é o mpg previsto associado a uma potência de 98? Quais são os intervalos de previsão e confiança 
de 95% associados?  

(b) Plote a resposta e o preditor. Use a função abline() para exibir a linha de regressão de mínimos quadrados. 

(c) Use a função plot() para produzir gráficos de diagnóstico do ajuste de regressão de mínimos quadrados.

Comente sobre quaisquer problemas que você vir com o ajuste. 

In [1]:
install.packages("ISLR")
library(ISLR)

Updating HTML index of packages in '.Library'
Making 'packages.html' ... done


In [4]:
lm.fit=lm(mpg~horsepower, data="Auto")

ERROR: Error in eval(predvars, data, env): invalid 'envir' argument of type 'character'


In [5]:
lm.fit


Call:
lm(formula = mpg ~ horsepower, data = Auto)

Coefficients:
(Intercept)   horsepower  
    39.9359      -0.1578  


Os resultados mostram o intercepto e o coeficiente beta para a variável horsepower

Interpretação da saída acima:

a equação da linha de regressão estimada pode ser escrita da seguinte forma: mpg = 39.94 + -0.16*horsepower

o intercepto(b0) é 39.94. Pode ser interpretado como a unidade de mpg(consumo de combustível) prevista para cada zero de horsepower. Lembre-se que estamos operando em unidades em milhas por galão. Isso significa que, para um intercepto  de horsepower igual a zero, podemos esperar um consumo de combustível de 39.94 * 1.000 = 39.940.



In [ ]:
# horsepower = preditor | mpg = resposta

In [6]:
summary(lm.fit)


Call:
lm(formula = mpg ~ horsepower, data = Auto)

Residuals:
     Min       1Q   Median       3Q      Max 
-13.5710  -3.2592  -0.3435   2.7630  16.9240 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 39.935861   0.717499   55.66   <2e-16 ***
horsepower  -0.157845   0.006446  -24.49   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 4.906 on 390 degrees of freedom
Multiple R-squared:  0.6059,	Adjusted R-squared:  0.6049 
F-statistic: 599.7 on 1 and 390 DF,  p-value: < 2.2e-16


In [9]:
confint(lm.fit)

2.5 %     97.5 %    
(Intercept) 38.525212 41.3465103
horsepower  -0.170517 -0.1451725